packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,297 @@
+    uint256 linearSecondsLeft = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+    require(
+      timeSinceL2 < linearSecondsLeft,
+      "Can only update fraction once block reward calculation for years 15-30 has been implemented."
+    );
+    require(partner != address(0), "Partner cannot be the zero address.");
+    require(
+      partner != carbonOffsettingPartner || value != carbonOffsettingFraction.unwrap(),
+      "Partner and value must be different from existing carbon offsetting fund."
+    );
+    require(value < FixidityLib.fixed1().unwrap(), "Value must be less than 1.");
+    carbonOffsettingPartner = partner;
+    carbonOffsettingFraction = FixidityLib.wrap(value);
+    require(
+      FixidityLib.newFixed(1).gte(communityRewardFraction.add(carbonOffsettingFraction)),
+      "Sum of partner fractions must be less than or equal to 1."
+    );
+    emit CarbonOffsettingFundSet(partner, value);
+    return true;
+  }
+
+  /**
+   * @notice Calculates remaining CELO balance to mint.
+   * @return The remaining CELO balance to mint.
+   */
+  function getRemainingBalanceToMint() public view returns (uint256) {
+    return GOLD_SUPPLY_CAP - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @return The total balance minted by the MintGoldSchedule contract.
+   */
+  function getTotalMintedBySchedule() public view returns (uint256) {
+    return totalMintedBySchedule;
+  }
+
+  /**
+   * @return The currently mintable amount.
+   */
+  function getMintableAmount() public view returns (uint256) {
+    (uint256 targetGoldTotalSupply, , ) = getTargetGoldTotalSupply();
+    return targetGoldTotalSupply - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @notice Returns the target CELO supply according to the target schedule.
+   * @return targetGoldTotalSupply The target total CELO supply according to the target schedule.
+   * @return communityTargetRewards The community reward that can be minted according to the target schedule.
+   * @return carbonFundTargetRewards The carbon offsetting reward that can be minted according to the target schedule.
+   */
+  function getTargetGoldTotalSupply()
+    public
+    view
+    whenDependenciesSet
+    returns (
+      uint256 targetGoldTotalSupply,
+      uint256 communityTargetRewards,
+      uint256 carbonFundTargetRewards
+    )
+  {
+    require(block.timestamp > genesisStartTime, "genesisStartTime has not yet been reached.");
+    require(block.timestamp > l2StartTime, "l2StartTime has not yet been reached.");
+
+    uint256 timeSinceL2 = block.timestamp - l2StartTime;
+    uint256 linearSecondsLeft = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+    uint256 mintedOnL1 = totalSupplyAtL2Start - GENESIS_GOLD_SUPPLY;
+
+    // Pay out half of all block rewards linearly.
+    uint256 l1LinearRewards = (GOLD_SUPPLY_CAP - GENESIS_GOLD_SUPPLY) / 2; //(200 million) includes validator rewards.
+    uint256 l2LinearRewards = l1LinearRewards - mintedOnL1;
```

Comment:
Can we double check that _l1LinearRewards_ cannot be > _mintedOnL1_

Reply:
On second though it should probably be fine since we have 15 years scheduled for linear rewards